apps/dashboard/features/governance/hooks/useProposals.ts

```diff
@@ -0,0 +1,351 @@
+  totalPages: number;
+  itemsPerPage: number;
+  currentItemsCount: number;
+}
+
+type Proposal = Omit<Query_Proposals_Items, "endBlock" | "startBlock">;
+
+// Helper function to transform GraphQL proposal data to governance component format
+const transformToGovernanceProposal = (
+  graphqlProposal: Proposal,
+): GovernanceProposal => {
+  const forVotes = parseInt(graphqlProposal.forVotes);
+  const againstVotes = parseInt(graphqlProposal.againstVotes);
+  const abstainVotes = parseInt(graphqlProposal.abstainVotes);
+  const quorum = parseInt(graphqlProposal.quorum);
+
+  const total = forVotes + againstVotes + abstainVotes;
+
+  const forPercentage = total > 0 ? Math.round((forVotes / total) * 100) : 0;
+  const againstPercentage =
+    total > 0 ? Math.round((againstVotes / total) * 100) : 0;
+
+  // Map GraphQL status to our enum
+  const getProposalStatus = (status: string): ProposalStatus => {
+    switch (status.toLowerCase()) {
+      case "active":
+        return ProposalStatus.ONGOING;
+      case "succeeded":
+      case "executed":
+        return ProposalStatus.EXECUTED;
+      case "defeated":
+        return ProposalStatus.DEFEATED;
+      case "cancelled":
+        return ProposalStatus.CANCELLED;
+      case "pending":
+        return ProposalStatus.PENDING;
+      default:
+        return ProposalStatus.PENDING;
+    }
+  };
+
+  const formatVotes = (votes: number): number => {
+    if (typeof votes !== "number" || isNaN(votes) || votes < 0) {
+      return 0;
+    }
+
+    try {
+      const formattedVotes = Number(formatEther(BigInt(Math.floor(votes))));
+      return formattedVotes;
+    } catch (error) {
+      console.warn("Error formatting votes:", votes, error);
+      return 0;
+    }
+  };
+
+  const getProposalState = (status: string): ProposalState => {
+    switch (status.toLowerCase()) {
+      case "active":
+        return ProposalState.ACTIVE;
+      case "succeeded":
+      case "executed":
+      case "defeated":
+      case "cancelled":
+        return ProposalState.COMPLETED;
+      case "pending":
+      default:
+        return ProposalState.WAITING_TO_START;
+    }
+  };
+
+  // Helper function to calculate days difference
+  const calculateDaysDiff = (
+    timestamp: string,
+    currentTime: number,
+  ): number => {
+    const diffSeconds = Math.abs(parseInt(timestamp) - currentTime);
+    return Math.floor(diffSeconds / (24 * 60 * 60));
+  };
+
+  const getTimeText = (startTimestamp: string, endTimestamp: string) => {
+    const now = Date.now() / 1000;
+    const startTime = parseInt(startTimestamp);
+    const endTime = parseInt(endTimestamp);
+
+    if (startTime > now) {
+      const days = calculateDaysDiff(startTimestamp, now);
+      return `${days}d to start`;
+    } else if (endTime > now) {
+      const days = calculateDaysDiff(endTimestamp, now);
+      return `${days}d left`;
+    } else {
+      const days = calculateDaysDiff(endTimestamp, now);
+      return `${days}d ago`;
+    }
```

Comment:
it would be cool to have the hours as well

<img width="474" height="82" alt="Image" src="https://github.com/user-attachments/assets/8c68acc9-0824-40fa-a08c-efae905ac3e1" />